src/ensembl/src/content/app/entity-viewer/gene-view/GeneView.tsx

```diff
@@ -361,7 +364,17 @@ const useGeneViewRouting = () => {
 
   useEffect(() => {
     if (view && viewInRedux !== view) {
-      dispatch(updateView(view as View));
+      if (ensureGeneViewExists(view)) {
```

Comment:
Let's try to get rid of the nested conditionals.

```ts
if (view && isViewParameterValid(valid) && viewInRedux !== view) {
  dispatch(updateView(view as View));
} else {
  // I believe the code as it stands will already do the right thing,
  // because initially the view in redux is "transcripts"
}
```

Reply:
✅